text/0000-sigstore-integration.md

```diff
@@ -0,0 +1,387 @@
+
+One of the first principles for considering the implementing this type of tooling in the community ecosystem is making sure that the developer experience remains a Tier 1 consideration. Historically, when security specific capabilities are brought into the mix the impact to productivity and developer experience has been of such a level that very often the security “stuff” gets ignored or just falls to the wayside (see PGP signing as an example). This is one area where Sigstore truly shines. From the outset Sigstore was designed to be a toolchain that would be a trivial impact to the day to day workflow of a developer as to be a non-issue. The cosign application is easy to use on the desktop (and fairly easy to drop into a build system)  and the integration with many widely used IdP implementations (GH, GCP, etc) make the workflow simple. There are no long-term keys to manage, nor any other infrastructure when you are utilizing the public service. Further the verification service (Rekor) is accessed through a well documented API making both desktop CLI and machine level access a straight-forward proposition.
+
+While not a part of this RFC, end users can implement additional projects such as [in-toto](https://in-toto.io/) to build more advanced verification and attestation of software.
+
+## Goals
+
+This proposal will enhance cargo and crates.io through adoption of the Sigstore capabilities and workflows for supporting both use cases as outlined above. Specifically, 
+
+* Adopt Sigstore infrastructure to allow for the signing of crates for use both above.
+* Facilitate workflows in both Cargo and crates.io to allow for signature generation in the former case and verification in the latter case of crates and their dependencies by crate consumers (a note on dependencies: the depth level should be 1 for verification in this first phase)
+
+This implementation will support all three Sigstore implementation models which will be effectively transparent to both crates.io maintainers and crates.io consumers. At a high level the three Sigstore implementation models are:
+
+1. Use of the public Sigstore common good service, signing with the Sigstore public root.
+1. Signature operations via a private fulcio instance with intermediate signer issued internally, but signature metadata pushed to public Rekor instance.
+1. Full internal Sigstore infra used with internal consumption only software.
+
+# Non Goals
+
+* Implementing TUF - this proposal describes a sequenced implementation with TUF in a phase 2. 
+* Signing crates.io index - this is covered by other proposals, but is discussed in the context of TUF later in this document.
```

Comment:
This seems to reflect the older git-based registry of crates.io and also would not be very possible under sparse registries, assuming that I'm interpreting this as signing the actual commit for the index itself.

I think that it's worth clarifying that, by index, you mean the effective state of crates.io, not simply ensuring that all crates are signed.